extensions.html

```diff
@@ -174,30 +176,55 @@ <h2><dfn>GamepadHand</dfn> Enum</h2>
     <section>
       <h2><dfn>GamepadHapticActuator</dfn> Interface</h2>
       <p>
-        Each <code>GamepadHapticActuator</code> corresponds to a motor or other
-        actuator that can apply a force for the purposes of haptic feedback.
+        A <code>GamepadHapticActuator</code> corresponds to a configuration of
+        motors or other actuators that can apply a force for the purposes of
+        haptic feedback.
       </p>
 
       <pre class="idl">
         interface GamepadHapticActuator {
           readonly attribute GamepadHapticActuatorType type;
-          Promise<void> pulse(double value, double duration);
+          Promise<GamepadHapticsResult> playEffect(
+              GamepadHapticEffectType type,
+              GamepadEffectParameters params);
+          Promise<GamepadHapticsResult> reset();
         };
       </pre>
 
       <dl data-dfn-for="GamepadHapticActuator">
-        <dt><dfn data-lt="pulse()">pulse</dfn></dt>
+        <dt><dfn data-lt="playEffect()">playEffect</dfn></dt>
         <dd>
           <p>
-            <code>pulse()</code> applies a value to the actuator for
-            <var>duration</var> milliseconds. The value passed to
-            <code>pulse()</code> is clamped to limits defined by the actuator
-            type. The returned Promise will resolve <code>true</code> once the
-            pulse has completed.
+            <code>playEffect()</code> interprets the effect parameters in
+            <var>params</var> as a haptic effect with the specified
+            <var>type</var> and issues it to the gamepad. It returns a
+            Promise that resolves to <code>'complete'</code> once the effect
+            is complete.
           </p>
           <p>
-            Repeated calls to <code>pulse()</code> override the previous
-            values.
+            If the effect parameters are invalid, <code>playEffect()</code>
+            must return with a Promise resolved to
+            <code>'invalid-parameter'</code>. The validity of the effect
+            parameters is defined by the effect type.
+          </p>
+          <p>
+            If the effect type is not compatible with the actuator,
+            <code>playEffect()</code> must immediately return a Promise resolved
```

Comment:
The use of the word "immediately" seems to imply that if I execute the following,

```
actuator.playEffect(invalid_params).then(() => {
  console.log("1");
});
Promise.resolve().then(() => {
  console.log("2");
});
```

Then it is guaranteed to print,

```
1
2
```

In my experience APIs returning Promises run the steps that result in the resolution of the promise "in parallel" which means that the exact timing of when the promise resolves is not directly tied to Javascript execution and could, for example, first require a round-trip to between a renderer and browser process before the result is known.

Reply:
I think this should be "in parallel". I removed the requirement that the promise is immediately resolved.